src/FacadeAccountCreation/FacadeAccountCreation.Core/Services/CompaniesHouse/CompaniesHouseLookupService.cs

```diff
@@ -17,7 +17,12 @@ public CompaniesHouseLookupService(HttpClient httpClient)
 
     public async Task<CompaniesHouseResponse?> GetCompaniesHouseResponseAsync(string id)
     {
-        var response = await _httpClient.GetAsync($"{CompaniesHouseEndpoint}/{id}");
+        var uriBuilder = new UriBuilder(_httpClient.BaseAddress)
```

Comment:
What's the advantage of using UriBuilder when the previous code is also correct & simpler?

Reply:
It's throwing "Change this code to not construct the URL from user-controlled data" issue from SonarQube10.4

Previous code leads to security vulnerabilities such as injection attacks to prevent this we should use safer methods to build URL.

Please have a look this documentation
https://vss-sonarqube-upgrade.azure.defra.cloud/project/issues?issueStatuses=OPEN%2CCONFIRMED&open=3b8efc25-d623-49e6-9c3e-a3a907fed910&id=epr-pom-web-api-gateway

@davidtonthat

Reply:
This looks reasonable. Please ensure manual testing passes.